[FLINK-33703] Use maven 3.8.6 for release profile

snuyanzin opened the pull request.

XComp commented on pom.xml:
```diff
@@ -133,8 +133,7 @@ under the License.
                                         <requireMavenVersion>
-                                            <!-- maven version must be lower than 3.3. See FLINK-3158 -->
-                                            <version>(,3.3)</version>
+                                            <version>[3.8.6]</version>
```
```suggestion
                                            <!-- Maven version for releasing should match the Maven version used in CI (and Flink's main repo) -->
                                            <version>[3.8.6]</version>
```
nit: can we add a bit more context to why this version is selected?

snuyanzin commented:
Addressed feedback
also noticed that latest jackson 2.18.2 requires maven-shade-plugin 3.6.0 since it has a version folder for java 22.
And maven-shade-plugin 3.6.0 requires at least maven 3.6.3
